### ISSUE TITLE
Cross Validation for P2

### ISSUE THREAD

**Issue by 161siegels** [Quantco/glum#772]
Do you have suggestions or recommendations for iterating through possibly P2 constructions? For instance, I am interested in the cross validation of regularization strengths on groups of parameters but this seems difficult as I currently understand since alpha (the parameter tuned in cv) is applied to all features. I know I can edit P2 but this is not part of the current CV framework. Maybe i am misunderstanding but if not, what do you recommend?

**Comment by lbittarello** [comment]
Have you considered using [scikit-learn's framework](https://scikit-learn.org/stable/modules/grid_search.html)? Glum is fully compatible with scikit-learn. :)

**Comment by MarcAntoineSchmidtQC** [comment]
The built-in cross-validation framework is only implemented for alpha because this is where we can take advantage of the regularization path. For P2, I recommend using sklearn's implementation, as @lbittarello suggested.

While I think P2 could also somewhat benefit from a form of regularization path, it is currently not planned to add this feature.
